sbin/prepareWorkspace.sh

```diff
@@ -51,29 +51,41 @@ copyFromDir() {
 # this is workarounding --strip-components 1 missing on gnu tar
 # it requires  absolute tar-filepath as it changes dir and is hardcoded to one
 # similar approach can be used also for zip in future
-untarGnuAbsPathWithStripComponents1() {
+unpackGnuAbsPathWithStripComponents1() {
   local tmp=$(mktemp -d)
   pushd "$tmp" > /dev/null
-    tar "$@"
+    "$@"
   popd  > /dev/null
   mv "$tmp"/*/* .
   mv "$tmp"/*/.* . || echo "no hidden files in tarball"
   rmdir "$tmp"/*
```

Comment:
There are directories, this will fail at least on Windows.

```suggestion
  rm -rf "$tmp"/*
  rm -rf "$tmp"
```

Reply:
I think this suggestion is wrong. It is intentional, that rmdir will fail if it is not empty.  If something remained here, then I think there was bug before. What directories remained here?  Was there something in console before?

Reply:
You're right. I've forgotten to update the `previous` lines. We can't move the files because it leads to:

`mv /tmp/tmp.iwVmIoZLAI/jdk-21.0.0+0-static-libs/lib .
mv: cannot move '/tmp/tmp.iwVmIoZLAI/jdk-21.0.0+0-static-libs/lib' to './lib': Directory not empty`

The update was from 

`````
  mv "$tmp"/*/* .
  mv "$tmp"/*/.* . || echo "no hidden files in tarball"
`````

to

`````
  cp -r "$tmp"/*/* .
  cp -r "$tmp"/*/.* . || echo "no hidden files in tarball"
`````

Whole suggestion looks like this:

`````
  cp -r "$tmp"/*/* .
  cp -r "$tmp"/*/.* . || echo "no hidden files in tarball"
  rm -rf "$tmp"/*
  rm -rf  "$tmp"
`````

Reply:
That is slightly different case. I do not want to mix those two changesets in this PR.

To change from `mv and rmdir` to `cp and rm -rf` really shoudl be separate change set. Maybe different function Aka  
```
// this method is very simialr to unpackGnuAbsPathWithStripComponents1
// however is able to mege archive even if targets exists, and rpevent
//mv /tmp/tmp.iwVmIoZLAI/jdk-21.0.0+0-static-libs/lib . mv: cannot move '/tmp/tmp.iwVmIoZLAI/jdk-21.0.0+0-static-libs/lib' to './lib': Directory not empty
// hoewer, is slightly unsafe. If you can, use unpackGnuAbsPathWithStripComponents1
unpackAndMErgeGnuAbsPathWithStripComponents1() {
  local tmp=$(mktemp -d)
  pushd "$tmp" > /dev/null
    "$@"
  popd  > /dev/null
  cp -ar "$tmp"/*/* .
  cp -ar "$tmp"/*/.* . || echo "no hidden files in tarball"
  rm -rf "$tmp"/*
  rm -rf "$tmp"
}
```